src/devices/bus/psx/parallel.h

```diff
@@ -0,0 +1,78 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#ifndef MAME_BUS_PSX_PARALLEL_H
+#define MAME_BUS_PSX_PARALLEL_H
+
+#pragma once
+
+//**************************************************************************
+//  INTERFACE CONFIGURATION MACROS
+//**************************************************************************
+
+#define MCFG_PSX_PARALLEL_SLOT_ADD(_tag, _slot_intf, _def_slot) \
+	MCFG_DEVICE_ADD(_tag, PSX_PARALLEL_SLOT, 0) \
+	MCFG_DEVICE_SLOT_INTERFACE(_slot_intf, _def_slot, false)
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> psx_parallel_slot_device
+
+class psx_parallel_interface;
+
+class psx_parallel_slot_device : public device_t, public device_slot_interface
+{
+public:
+	// construction/destruction
+	psx_parallel_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+	virtual ~psx_parallel_slot_device();
+
+	DECLARE_READ16_MEMBER(exp_r);
+	DECLARE_WRITE16_MEMBER(exp_w);
+
+	bool hascard()
+	{
+		if (m_card)
+			return true;
+		else return false;
```

Comment:
Inconsistent line breaks.  Also, this could be simplified to `return bool(m_card);` or `return nullptr != m_card;` and the method should be `const`-qualified since it isn't a mutator.